src/ChannelServer/World/Monster.cs

```diff
@@ -149,5 +228,157 @@ public void Kill(Character killer)
 
 			Send.ZC_DEAD(this);
 		}
+
+		/// <summary>
+		/// Attack target entity if monster is agred
+		/// </summary>
+		public bool IsTargetVisible()
+		{
+			if (this.Target.Position.InRange2D(
+				this.Position, World.Map.VisibleRange
+			))
+				return true;
+			else
+			{
+				this.Target = null;
+				return false;
+			}
+		}
+
+		/// <summary>
+		/// Sets direction and updates clients.
+		/// </summary>
+		/// <param name="d1"></param>
+		/// <param name="d2"></param>
+		public void Rotate(float d1, float d2)
+		{
+			if (this.Hp <= 0) return;
+
+			this.SetDirection(d1, d2);
+			Send.ZC_ROTATE(this);
+		}
+		
+		/// <summary>
+		/// Sets direction toward target and updates clients.
+		/// </summary>
+		public void RotateTowardTarget()
+		{
+			if (this.Hp <= 0) return;
+
+			if (this.Target == null) return;
+			this.SetDirection(
+				Shared.Util.Math2.AngleBetweenTwoEntity(
```

Comment:
Note: Personally I believe a `using` increases the readability significantly, because you don't need the full name. Even if it's only for one two instances.